gms/gossiper.cc

```diff
@@ -337,27 +341,28 @@ future<> gossiper::handle_ack_msg(msg_addr id, gossip_digest_ack ack_msg) {
                                     from, p.ack_msg_digest, p.pending);
                             ack_msg_digest = std::move(p.ack_msg_digest.value());
                             p.ack_msg_digest= {};
-                            return stop_iteration::no;
+                            continue;
                         } else {
                             // No more pending ack msg digests to process
                             p.pending = false;
                             logger.debug("No more queued gossip ack msg digests from node {}, ack_msg_digest={}, pending={}",
                                     from, p.ack_msg_digest, p.pending);
-                            return stop_iteration::yes;
+                            co_return;
                         }
-                    }).handle_exception([this, from] (std::exception_ptr ep) {
+        } catch (...) {
+            ep = std::current_exception();
                         if (_ack_handlers.contains(from.addr)) {
                             ack_msg_pending& p = _ack_handlers[from.addr];
                             p.pending = false;
                             p.ack_msg_digest = {};
                             logger.warn("Failed to process gossip ack msg digests from node {}: {}", from, ep);
                         }
-                        return make_exception_future<stop_iteration>(ep);
```

Comment:
Similarly, here. Can't ep live in the catch clause? Makes everything simpler to read later.